haskell.nix/flake.nix

```diff
@@ -0,0 +1,116 @@
+{
+  inputs = {
+    flake-compat = {
+      url = "github:edolstra/flake-compat";
+      flake = false;
+    };
+    nixpkgs.url = "github:serokell/nixpkgs";
+    haskell-nix.url = "github:serokell/haskell.nix/serokell-latest";
+    hackage-nix = {
+      url = "github:input-output-hk/hackage.nix";
+      flake = false;
+    };
+    stackage-nix = {
+      url = "github:input-output-hk/stackage.nix";
+      flake = false;
+    };
+
+    # if you want to run weeder:
+    ##haskell-nix-weeder = {
+    ##  url = "github:serokell/haskell-nix-weeder";
+    ##  flake = false;
+    ##};
+  };
+
+  outputs = { self, nixpkgs, haskell-nix, hackage-nix, stackage-nix, ... }:
+
+  # for weeder:
+  ##outputs = { self, nixpkgs, haskell-nix, hackage-nix, stackage-nix, haskell-nix-weeder, ... }:
+
+    let
+      haskellNix = import haskell-nix {
+        sourcesOverride = { hackage = hackage-nix; stackage = stackage-nix; };
+      };
+
+      pkgs = import nixpkgs haskellNix.nixpkgsArgs;
+      lib = pkgs.lib;
+
+      # invoke haskell.nix
+      hs-pkgs = pkgs.haskell-nix.stackProject {
+        src = pkgs.haskell-nix.haskellLib.cleanGit {
+          name = "pataq-package";
```

Comment:
Perhaps

```suggestion
          name = throw "Your package name goes here";
```

Reply:
I'm not sure about this, people would need to guess to remove `throw` here, and in other places, where `pataq-package` is an attrset key, we can't replace it with a throw anyway.

Reply:
It does seem nice in principle to not let it evaluate until every name has been updated, but I still agree with @zhenyavinogradov